debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/legacy/SnapshotReaderIT.java

```diff
@@ -351,16 +351,15 @@ public void shouldCreateSnapshotSelectively() throws Exception {
     }
 
     @Test
-    public void shouldCreateSnapshotOfSingleDatabaseUsingReadEvents() throws Exception {
+    public void shouldCreateSnapshotOfSingleDatabaseUsingInsertEvents() throws Exception {
         config = simpleConfig()
                 .with(MySqlConnectorConfig.DATABASE_INCLUDE_LIST, "connector_(.*)_" + DATABASE.getIdentifier())
-                .with(MySqlConnectorConfig.SNAPSHOT_EVENTS_AS_INSERTS, false)
```

Comment:
The SMT should be applied here?

Reply:
As `SnapshotReaderIT` does not use the connector for testing so we can't apply the SMT through config properties instead should be applied manually. So as per @jpechane suggestion, this test should be left untouched and a snapshot is emitted using `generateInsertEvents()`.

Reply:
Ah, I see. So I think this entire test method is obsolete then and can be removed. If there's no actual implementation that emits snapshot events as inserts, there's no point in testing it (or having a method on the reader solely for the purpose of testing it). And we do test the SMT approach with the new implementation.

Reply:
@gunnarmorling Also, every test in `SnapshotReaderIT` is using `generateInsertEvents()` as it was the default behavior of the connector. Hence we need to convert them `generateReadEvents()` now as we are removing the implementation of `generateInsertEvents()`. Wdyt?

Reply:
Yes, exactly. `generateReadEvents()` should be used, and affected tests should be adjusted accordingly.